backend/lambdas/process_post/lambda.py

```diff
@@ -0,0 +1,193 @@
+    extracted_info.location = extracted_info.location if extracted_info.location != "<UNKNOWN>" else ""
+    extracted_info.sentiment = extracted_info.sentiment if extracted_info.sentiment != "<UNKNOWN>" else ""
+
+
+def text_topic_match(
+        meta_topics: str,
+        text: str,
+) -> TopicMatch:
+
+    bedrock_llm = ChatBedrock(
+        model_id=MODEL_ID,
+        model_kwargs=TOPIC_MATCH_MODEL_PARAMETERS,
+        client=bedrock_runtime,
+    )
+
+    claude_topic_match_prompt_template = TOPIC_MATCH_PROMPT_SELECTOR.get_prompt(MODEL_ID)
+
+    print(claude_topic_match_prompt_template.format(meta_topics=meta_topics, text=text))
+
+    structured_llm = bedrock_llm.with_structured_output(TopicMatch)
+
+    structured_topic_match_chain = claude_topic_match_prompt_template | structured_llm
+
+    topic_match_obj = structured_topic_match_chain.invoke({"meta_topics": meta_topics, "text": text})
+
+    print("Topic match object")
+    print(type(topic_match_obj))
+    print(topic_match_obj)
+
+    return topic_match_obj
+
+def text_information_extraction(
+        sentiments: str,
+        text: str
+) -> ExtractedInformation:
+
+    bedrock_llm = ChatBedrock(
+        model_id=MODEL_ID,
+        model_kwargs=INFORMATION_EXTRACTION_MODEL_PARAMETERS,
+        client=bedrock_runtime,
+    )
+
+    claude_information_extraction_prompt_template = INFORMATION_EXTRACTION_PROMPT_SELECTOR.get_prompt(MODEL_ID)
+
+    print("The prompt template")
+    print(claude_information_extraction_prompt_template)
+
+    print(claude_information_extraction_prompt_template.format(
+        text=text,
+        sentiments=sentiments
+        )
+    )
+
+    structured_llm = bedrock_llm.with_structured_output(ExtractedInformation)
+
+    structured_chain = claude_information_extraction_prompt_template | structured_llm
+
+    information_extraction_obj = structured_chain.invoke({
+        "text": text,
+        "sentiments": sentiments
+    })
+
+    print("Information extraction object")
+    print(type(information_extraction_obj))
+    print(information_extraction_obj)
+
+    return information_extraction_obj
+
+@logger.inject_lambda_context(log_event=True)
+def handler(event, _context: LambdaContext):
+
+        item = event
+
+        logger.info('Item:')
+        logger.info(item)
+
+        #Attemp to categorize item
+        text = item['text']
+        logger.info(f'Text: {text}')
+
+        text = demoji.replace(text, "")
+        item['text_clean'] = text
+
+        try:
+
+            #meta_topics_str = ','.join(META_TOPICS)
+            topic_match = text_topic_match(META_TOPICS_STR, item['text_clean'])
+
+            if topic_match.is_match == True and len(topic_match.related_topics) > 0:
+
+                try:
+
+                    text_insight = text_information_extraction(META_SENTIMENTS_STR, item['text_clean'])
+                    logger.info(f'Text insights:')
+                    logger.info(text_insight)
```

Comment:
You seem to be double-logging some things, for example line 119 logs the same as this line 150.

Reply:
Fixed in [e91f7a0](https://github.com/aws-samples/ai-powered-text-insights/pull/3/commits/e91f7a05ccb41059a49562d086e6824bc3af9ecc)